lib/chef/resource/execute.rb

```diff
@@ -143,6 +143,30 @@ def user(arg = nil)
         )
       end
 
+      def domain(arg = nil)
+        set_or_return(
+          :domain,
+          arg,
+          :kind_of => [ String ]
+        )
+      end
+
+      def password(arg = nil)
+        set_or_return(
+          :password,
+          arg,
+          :kind_of => [ String ]
+        )
+      end
+
+      def sensitive(args = nil)
+        if ! password.nil?
```

Comment:
actually i think you can do `property :password, String, sensitive: true` and avoid this.

the domain should also be a property (you don't have to convert the whole file, but anything new should be a property).